src/string-class.js

```
@@ -1,3 +1,104 @@
 'use strict';
 
-// TODO: implement String class extensions
+/**
+ * Returns true if the string contains any vowels
+ * @returns {Boolean}
+ */
+String.prototype.hasVowels = function() {
+  // new RegExp('[aeiou]', 'gi')
+  return /[aeiou]/gi.test(this);
+};
+
+/**
+ * Returns the string with all characters in upper case
+ * @returns {String}
+ */
+String.prototype.toUpper = function() {
+  return this.replace(/([a-z])/g, (match) => {
+    return String.fromCharCode(match.charCodeAt(0) - 32);
+  });
+};
+
+/**
+ * Returns the string with all characters in lower case
+ * @returns {String}
+ */
+String.prototype.toLower = function() {
+  return this.replace(/([A-Z])/g, (match) => {
+    return String.fromCharCode(match.charCodeAt(0) + 32);
+  });
+};
+
+/**
+ * Returns the string with the fisrt character capitalized
+ * @returns {String}
+ */
+String.prototype.ucFirst = function() {
+  return this.replace(/(^[a-z])/, (match) => {
+    return match.toUpper();
+  });
+};
+
+/**
+ * Returns true if the string is a question i.e it ends with a question mark
+ * @returns {Boolean}
+ */
+String.prototype.isQuestion = function() {
+  return /\?$/.test(this);
+};
+
+/**
+ * Returns a list of the words in the string as an array
+ * @returns {Array}
+ */
+String.prototype.words = function() {
+  return this.match(/[\w-']+/g);
+};
+
+/**
+ * Returns the number of words in the string
+ * @returns {Number}
+ */
+String.prototype.wordCount = function() {
+  return this.words().length;
+};
+
+/**
+ * Returns a currency representation of the string
+ * @returns {String}
+ */
+String.prototype.toCurrency = function() {
+  // if the string contains anything that's not a digit, comma or period, it
+  // cannnot be represented as a currency string
+  if(this.match(/[^(\d,\.)]/)){
+    throw(`${this} is not convertible into a currency string`);
```

I think doing that would make the `String#toCurrency` method seem like it's failing silently. The users would have to check whether the input string is equal to the output string after every call to `String#toCurrency`. Furthermore, I think we are assuming that `String#toCurrency` can only be used on strings composed of numbers, in which case it would make more sense to return `NaN` for failing strings.